web_widget_one2many_product_picker/models/product_product.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 Tecnativa - Alexandre D. Díaz
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+from odoo import api, fields, models, tools
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    image_variant_medium = fields.Binary(
+        "Variant Image Medium (Computed)",
+        compute='_compute_variant_image',
+        help="This field holds the image used as image for the product variant"
+             "or product image medium, limited to 512x512px.",
+    )
+
+    image_variant_big = fields.Binary(
+        "Variant Image Big (Computed)",
+        compute='_compute_variant_image',
+        help="This field holds the image used as image for the product variant"
+             "or product image, limited to 1024x1024px.",
+    )
+
+    @api.depends('image_variant', 'product_tmpl_id.image')
+    def _compute_variant_image(self):
+        self.ensure_one()
```

Comment:
You have removed `@api.one`, but this method can be accessed by multi-recordset. You can't restrict it this way, but do here the `for record in self` loop.